### PR TITLE
Add possibility to dely some of required jobs by marking them as always_run: false

### PR DESCRIPTION
**Pull request by jmguzik**

### CODE REVIEW

File: pkg/prowgen/prowgen.go [hunk 1]
```diff
@@ -193,14 +193,18 @@ func generatePresubmitForTest(jobBaseBuilder *prowJobBaseBuilder, name string, i
 
 	shortName := info.TestName(name)
 	base := jobBaseBuilder.Rehearsable(!opts.disableRehearsal).Build(jc.PresubmitPrefix)
+	rerunCommand := prowconfig.DefaultRerunCommandFor(shortName)
+	if opts.defaultDisable {
```
Comment (smg247) [inline pkg/prowgen/prowgen.go:197]: don't we also want to verify that the job isn't `optional` here?

File: pkg/prowgen/prowgen.go [hunk 2]
```diff
@@ -193,14 +193,18 @@ func generatePresubmitForTest(jobBaseBuilder *prowJobBaseBuilder, name string, i
 
 	shortName := info.TestName(name)
 	base := jobBaseBuilder.Rehearsable(!opts.disableRehearsal).Build(jc.PresubmitPrefix)
+	rerunCommand := prowconfig.DefaultRerunCommandFor(shortName)
```
Comment (smg247) [inline pkg/prowgen/prowgen.go:196]: I'm on the fence about this change, but you could also put the logic inside `prowconfig.DefaultRerunCommandFor`. This is the only usage.

Comment (jmguzik) [reply pkg/prowgen/prowgen.go:196]: Unfortunately `prowconfig.DefaultRerunCommandFor` is upstream

Comment (smg247) [reply pkg/prowgen/prowgen.go:196]: ah, it is. makes sense then.

**Comment by jmguzik** [comment]
Interesting, after closing and reopening, old test result for e2e is qualified as pass while others are retested with the always run setting to false. It seems to be a little buggy and misleading solution.

File: pkg/prowgen/prowgen.go (hunk 1 above)
Comment (jmguzik) [reply pkg/prowgen/prowgen.go]: conditions updated

**Comment by jmguzik** [comment]
breaking changes can be fixed by periodic

**Pull request merged**